game/gameboard.py

```diff
@@ -249,7 +248,12 @@ def _validate_matrix(self, matrix_to_test):
                     pass
```

Comment:
The `if entry in valid_entries` part could be decoupled into a `_is_valid_cell_value(entry)` method. You can also change the code to be:

```
if not self._is_valid_cell_value(entry)
    return False
```